bundles/org.openhab.binding.evcc/src/main/java/org/openhab/binding/evcc/internal/handler/EvccBaseThingHandler.java

```diff
@@ -258,9 +260,22 @@ protected void sendCommand(String url) {
                 ContentResponse response = httpClient.newRequest(url).timeout(5, TimeUnit.SECONDS)
                         .method(HttpMethod.POST).header(HttpHeader.ACCEPT, "application/json").send();
 
-                if (response.getStatus() != 200) {
-                    updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
-                    logger.warn("evcc API error: HTTP {}", response.getStatus());
+                if (response.getStatus() == 200) {
+                    logger.debug("Sending command was successful");
+                } else {
+                    @Nullable
+                    JsonObject responseJson = gson.fromJson(response.getContentAsString(), JsonObject.class);
+                    Optional.ofNullable(responseJson).ifPresent(json -> {
+                        if (json.has("error")) {
+                            logger.debug("Sending command was unsuccessful, got this error:\n {}",
+                                    responseJson.get("error").getAsString());
+                            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR,
+                                    responseJson.get("error").getAsString());
+                        } else {
+                            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR);
+                            logger.warn("evcc API error: HTTP {}", response.getStatus());
```

Comment:
Setting the state to offline is a bit premature. What if it is a transient error like a timeout or just a bad request, while update polling still works. I would leave it to the caller (child class) to determine what to do. From this method, you can only signal that there is something wrong. (maybe by exception and/or http status nu,ber)